src/components/Design.vue

```diff
@@ -0,0 +1,341 @@
+
+interface ProductItem {
+  name: string;
+}
+
+type Nullable<T> = T | null | undefined;
+type RuleOutcome = string | true;
+type RuleHandler = (any) => RuleOutcome;
+type NextHandler = (to?: RawLocation | false | Function | void) => void;
+
+export default Vue.extend({
+  name: "Design",
+  components: {
+    NewProject,
+    NewOrganism,
+    NewModel
+  },
+  data() {
+    return {
+      isValid: false,
+      project: null as Nullable<ProjectItem>,
+      projectRules: [v => !!v || "Project is required"] as ReadonlyArray<
+        RuleHandler
+      >,
+      isProjectCreationDialogVisible: false,
+      organism: null as Nullable<OrganismItem>,
+      organismRules: [v => !!v || "Organism is required"] as ReadonlyArray<
+        RuleHandler
+      >,
+      isOrganismCreationDialogVisible: false,
+      product: null as Nullable<ProductItem>,
+      productRules: [v => !!v || "Product is required"] as ReadonlyArray<
+        RuleHandler
+      >,
+      productOptions: [] as ProductItem[],
+      isAerobic: false,
+      showAdvanced: false,
+      bigg: false,
+      rhea: true,
+      model: null as Nullable<ModelItem>,
+      modelOptions: [] as ModelItem[],
+      isModelCreationDialogVisible: false,
+      maxPredictions: 3,
+      isSubmitted: false,
+      hasSubmissionError: false
+    };
+  },
+  computed: {
+    projectOptions(): ProjectItem[] {
+      return this.$store.state.projects.projects;
+    },
+    // TODO (Moritz Beber): Selection of organism in future should depend on the
+    //  selected project, i.e., you can only select public organisms and
+    //  organisms within that project.
+    organismOptions(): OrganismItem[] {
+      return this.$store.state.organisms.organisms;
+    },
+    allModels(): ModelItem[] {
+      return this.$store.state.models.models;
+    },
+    predictionRules(): RuleHandler[] {
+      const rules: RuleHandler[] = [];
+      rules.push((v: number) => {
+        return (
+          (v > 0 && v <= process.env.VUE_APP_MAX_PREDICTIONS) ||
+          `Must be between 0 and ${process.env.VUE_APP_MAX_PREDICTIONS}.`
+        );
+      });
+      return rules;
+    }
+  },
+  watch: {
+    project(project: ProjectItem): void {
+      this.selectModels(project, this.organism);
+    },
+    organism(organism: OrganismItem): void {
+      this.selectModels(this.project, organism);
+    }
+  },
+  methods: {
+    onNewProject(project: ProjectItem): void {
+      this.project = project;
+    },
+    onNewOrganism(organism: OrganismItem): void {
+      this.organism = organism;
+    },
+    fetchProducts(): void {
+      axios
+        .get(`${settings.apis.metabolicNinja}/products`)
+        .then((response: AxiosResponse<ProductItem[]>) => {
+          this.productOptions = response.data;
+        })
+        .catch((error: Error) => {
+          this.$store.commit("setFetchError", error, { root: true });
+        });
+    },
+    selectModels(project: ProjectItem, organism: OrganismItem): void {
```

Comment:
This didn't seem to work for me?

![Screenshot 2019-05-10 at 13 26 04](https://user-images.githubusercontent.com/8027271/57524130-2ef15f00-7327-11e9-875f-90210565220f.png)